openstack/container/experimental/capsules/results.go

```diff
@@ -0,0 +1,77 @@
+package capsules
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract is a function that accepts a result and extracts a capsule resource.
+func (r commonResult) Extract() (*Capsule, error) {
+	var s *Capsule
+	err := r.ExtractInto(&s)
+	return s, err
+}
+
+// GetResult represents the result of a get operation.
+type GetResult struct {
+	commonResult
+}
+
+// Represents a Container Orchestration Engine Bay, i.e. a cluster
+type Capsule struct {
+	// UUID for the capsule
+	UUID string `json:"uuid"`
+
+	// ID for the capsule
+	ID int `json:"id"`
+
+	// User ID for the capsule
+	UserID string `json:"user_id"`
+
+	// Project ID for the capsule
+	ProjectID string `json:"project_id"`
+
+	// cpu for the capsule
+	CPU float64 `json:"cpu"`
+
+	// Memory for the capsule
+	Memory string `json:"memory"`
+
+	// The name of the capsule
+	Name string `json:"meta_name"`
+
+	// Indicates whether capsule is currently operational. Possible values include:
+	// Running,
+	Status string `json:"status"`
+
+	// The created time of the capsule.
+	Created string `json:"created_at"`
+
+	// The updated time of the capsule.
+	Updated string `json:"updated_at"`
+
+	// Links includes HTTP references to the itself, useful for passing along to
+	// other APIs that might want a server reference.
+	Links []interface{} `json:"links"`
+
+	// The capsule version
+	CapsuleVersion string `json:"capsule_version"`
+
+	// The capsule restart policy
+	RestartPolicy string `json:"restart_policy"`
+
+	// The capsule metadata labels
+	MetaLables map[string]string `json:"meta_labels"`
+
+	// The list of containers uuids inside capsule.
+	ContainersUUIDs []string `json:"containers_uuids"`
+
+	// The capsule IP addresses
+	Addresses map[string]interface{} `json:"addresses"`
```

Comment:
Do the keys to `Addresses` change frequently or do they have a common structure to them? In your test fixture, you have:

```json
			"addresses": {
				"b1295212-64e1-471d-aa01-25ff46f9818d": [
					{
						"version": 4,
						"preserve_on_delete": false,
						"addr": "172.24.4.11",
						"port": "8439060f-381a-4386-a518-33d5a4058636"
					}
				]
			},
```

So instead of `map[string]interface{}`, would it be better to have the following?

```go
Addresses map[string]Address `json:"addresses"`
```

And then:

```go
type Address struct {
  PreserveOnDelete bool `json:"preserve_on_delete"`
  Addr string `json:"addr"`
  Port string `json:"port"`
  Version int `json:"version"`
}
```

If so, please link to the appropriate Zun python code where the Address struct is defined. It looks like this feature was recently added.